ci/scripts/deterministic-e2e-test.sh

```diff
@@ -9,26 +9,28 @@ echo "--- Download artifacts"
 buildkite-agent artifact download risingwave_simulation .
 chmod +x ./risingwave_simulation
 
-export RUNNER=./risingwave_simulation
-export RUST_LOG=off
+export RUST_LOG=info
+export LOGDIR=.risingwave/log
+
+mkdir -p $LOGDIR
 
 echo "--- deterministic simulation e2e, ci-3cn-1fe, ddl"
-seq 16 | parallel MADSIM_TEST_SEED={} $RUNNER './e2e_test/ddl/\*\*/\*.slt'
+seq 16 | parallel MADSIM_TEST_SEED={} './risingwave_simulation ./e2e_test/ddl/\*\*/\*.slt > $LOGDIR/ddl-{}.log && rm $LOGDIR/ddl-{}.log'
```

Comment:
I think the CI will not stop even if there is error. So we do not need to `rm`. Plus, each log name should be unique.

Reply:
The command after `&&` won't be run if the former returns an error.

Reply:
Check out the new example: https://buildkite.com/risingwavelabs/pull-request/builds/11034#01841d9a-9d82-49b9-9b45-38a15a305f50

Reply:
@wangrunji0408 Quite curious, how come the rest of the tests do not run? Seems that previously, the rest of the tests for the seeds would still run even if there is an error. [See e.g. ](https://buildkite.com/risingwavelabs/pull-request/builds/10900#0184176c-7b19-463c-a13e-79d09d9debb3)

Furthermore, how come the remainder of the tests run for recovery but not for e2e?

Reply:
> Quite curious, would all of the seeds still run even if one of them have error?

Ya, that's the behavior of GNU parallel.

> Furthermore, how come the remainder of the test cases run for recovery but not for e2e?

We don't have this line in the recovery test script:
```bash
# Exits as soon as any line fails.
set -euo pipefail
```

The reason is that currently recovery test is expected to fail due to some bugs. Showing the results for each test case can help us reveal problems in our system.